README.md

```diff
@@ -1,112 +1,53 @@
+Declare the plugin in your Nextflow pipeline configuration file:
 
-- `plugins/nf-hello/src/main` 
-
-    The plugin implementation sources.
-
-- `plugins/nf-hello/src/test` 
-
-    The plugin unit tests. 
-
-## Plugin classes
-
-- `HelloConfig`: simple example how to handle configuration options provided via the Nextflow configuration file. 
-
-- `HelloExtension`: show how create an extension class that can be used to create custom channel factories, operation and fuctions that can be imported in the pipeline script as DSL extensions.
-
-- `CO2FootprintFactory` and `CO2FootprintObserver`: show how to intercept workflow runtime events and react correspondly with custom code.
-
-- `HelloPlugin`: the plugin entry point.
-
-
-## Unit testing 
-
-Run the following command in the project root directory (ie. where the file `settings.gradle` is located):
-
-```bash
-./gradlew check
+```groovy title="nextflow.config"
+plugins {
+  id 'nf-co2footprint'
+}
 ```
 
-## Testing and debugging
-
-To run and test the plugin in for development purpose, configure a local Nextflow build with the following steps:
-
-1. Clone the Nextflow repository in your computer into a sibling directory:
-    ```bash
-    git clone --depth 1 https://github.com/nextflow-io/nextflow ../nextflow
-    ```
-  
-2. Configure the plugin build to use the local Nextflow code:
-    ```bash
-    echo "includeBuild('../nextflow')" >> settings.gradle
-    ```
-  
-   (Make sure to not add it more than once!)
-
-3. Compile the plugin alongside the Nextflow code:
-    ```bash
-    make compile
-    ```
-
-4. Run Nextflow with the plugin, using `./launch.sh` as a drop-in replacement for the `nextflow` command, and adding the option `-plugins nf-hello` to load the plugin:
-    ```bash
-    ./launch.sh run nextflow-io/hello -plugins nf-hello
-    ```
+This is all that is needed - Nextflow will automatically fetch the plugin code at run time.
 
-## Testing without Nextflow build
+Additionally, you can adjust the nf-co2footprint plugin parameters in your config file as follows:
 
-The plugin can be tested without using a local Nextflow build using those steps:
-
-1. generate required artifacts with `make buildPlugins`
-2. copy build/plugins/your-plugin to `$HOME/.nextflow/plugins`
-3. create a pipeline with your plugin and see in action via `nextflow run ./my-pipeline-script.nf`
-
-## Package, upload and publish
-
-The project should be hosted in a GitHub repository whose name should match the name of the plugin, that is the name of the directory in the `plugins` folder (e.g. `nf-hello`).
-
-Follow these steps to package, upload and publish the plugin:
+```groovy title="nextflow.config"
+co2footprint {
+    file        = "${params.outdir}/co2footprint.txt"
+    reportFile  = "${params.outdir}/co2footprint_report.html"
+    ci          = 300
+    pue         = 1.4
+    country     = 'Germany'
+}
+```
```

Comment:
I would add an example of how to provide these file in a nextflow run:
`nextflow run hello -c nextflow.config`